client/core/simnet_trade.go

```diff
@@ -877,154 +849,173 @@ func testOrderStatusReconciliation(s *simulationTest) error {
 		return fmt.Errorf("client 2 locked funds not returned: locked before trading %v, locked after trading %v, "+
 			"locked after reconnect %v", preTradeLockedBalance, preDisconnectLockedBalance, c2Balance.Locked)
 	}
+	s.client2.enableWallets()
+
+	for _, c := range s.clients {
+		if err := c.mineMedian(context.TODO(), s.quote.id); err != nil {
+			return err
+		}
+		if err := c.mineMedian(context.TODO(), s.base.id); err != nil {
+			return err
+		}
+	}
+
+	s.waitALittleBit()
+
 	return nil
 }
 
-// TestResendPendingRequests runs a simple trade test, simulates init/redeem
-// request errors during trade negotiation and ensures that failed requests
-// are retried and the trades complete successfully.
-func testResendPendingRequests(s *simulationTest) error {
-	var qty, rate uint64 = 1 * s.lotSize, 250 * s.rateStep
-	s.client1.isSeller, s.client2.isSeller = true, false
+var _ comms.WsConn = (*tConn)(nil)
 
-	c1OrderID, c2OrderID, err := s.placeTestOrders(qty, rate)
+type tConn struct {
+	routeCh      chan string
+	errCh        chan error
+	on           uint32
+	closeAndWait func()
+
+	realConn comms.WsConn
+}
+
+func (tc *tConn) NextID() uint64 {
+	return tc.realConn.NextID()
+}
+func (tc *tConn) IsDown() bool {
+	return tc.realConn.IsDown()
+}
+func (tc *tConn) Send(msg *msgjson.Message) error {
+	return tc.realConn.Send(msg)
+}
+func (tc *tConn) Request(msg *msgjson.Message, respHandler func(*msgjson.Message)) error {
+	tc.routeCh <- msg.Route
+	err := <-tc.errCh
 	if err != nil {
 		return err
 	}
-
-	// Monitor trades and stop at order.MakerSwapCast.
-	monitorTrades, ctx := errgroup.WithContext(context.Background())
-	monitorTrades.Go(func() error {
-		return s.monitorOrderMatchingAndTradeNeg(ctx, s.client1, c1OrderID, order.MakerSwapCast)
-	})
-	monitorTrades.Go(func() error {
-		return s.monitorOrderMatchingAndTradeNeg(ctx, s.client2, c2OrderID, order.MakerSwapCast)
-	})
-	if err = monitorTrades.Wait(); err != nil {
+	return tc.realConn.Request(msg, respHandler)
```

Comment:
The MITM approach is great.   I think we should have done this from the start.  :hearts: